### ISSUE TITLE
TypeError: descriptor 'date' of 'datetime.datetime' object needs an argument

### ISSUE THREAD

**Issue by inducer** [pimutils/khal#707]
I get this traceback:

```
Traceback (most recent call last):i Walker (*1979) ⟳
  File "/home/andreas/src/env-3.6/lib/python3.6/site-packages/khal/ui/__init__.py", line 1300, in start_pane
    loop.run()
  File "/home/andreas/src/env-3.6/lib/python3.6/site-packages/urwid/main_loop.py", line 278, in run
    self._run()
  File "/home/andreas/src/env-3.6/lib/python3.6/site-packages/urwid/main_loop.py", line 376, in _run
    self.event_loop.run()
  File "/home/andreas/src/env-3.6/lib/python3.6/site-packages/urwid/main_loop.py", line 682, in run
    self._loop()
  File "/home/andreas/src/env-3.6/lib/python3.6/site-packages/urwid/main_loop.py", line 719, in _loop
    self._watch_files[fd]()
  File "/home/andreas/src/env-3.6/lib/python3.6/site-packages/urwid/raw_display.py", line 393, in <lambda>
    event_loop, callback, self.get_available_raw_input())
  File "/home/andreas/src/env-3.6/lib/python3.6/site-packages/urwid/raw_display.py", line 493, in parse_input
    callback(processed, processed_codes)
  File "/home/andreas/src/env-3.6/lib/python3.6/site-packages/urwid/main_loop.py", line 403, in _update
    self.process_input(keys)
  File "/home/andreas/src/env-3.6/lib/python3.6/site-packages/urwid/main_loop.py", line 503, in process_input
    k = self._topmost_widget.keypress(self.screen_size, k)
  File "/home/andreas/src/env-3.6/lib/python3.6/site-packages/urwid/wimp.py", line 643, in keypress
    return self._current_widget.keypress(size, key)
  File "/home/andreas/src/env-3.6/lib/python3.6/site-packages/urwid/container.py", line 1128, in keypress
    return self.body.keypress( (maxcol, remaining), key 
  File "/home/andreas/src/env-3.6/lib/python3.6/site-packages/khal/ui/__init__.py", line 1106, in keypress
    return super().keypress(size, key)
  File "/home/andreas/src/env-3.6/lib/python3.6/site-packages/khal/ui/base.py", line 82, in keypress
    return super().keypress(size, key)
  File "/home/andreas/src/env-3.6/lib/python3.6/site-packages/khal/ui/widgets.py", line 308, in keypress
    key = super(NextMixin, self).keypress(size, key)
  File "/home/andreas/src/env-3.6/lib/python3.6/site-packages/urwid/container.py", line 2269, in keypress
    key = w.keypress((mc,) + size[1:], key)
  File "/home/andreas/src/env-3.6/lib/python3.6/site-packages/khal/ui/__init__.py", line 919, in keypress
    self.edit(self.focus_event.event)
  File "/home/andreas/src/env-3.6/lib/python3.6/site-packages/khal/ui/__init__.py", line 756, in edit
    editor = EventEditor(self.pane, event, update_colors, always_save=always_save)
  File "/home/andreas/src/env-3.6/lib/python3.6/site-packages/khal/ui/editor.py", line 342, in __init__
    self.event.recurobject, self._conf, event.start_local,
  File "/home/andreas/src/env-3.6/lib/python3.6/site-packages/khal/ui/editor.py", line 569, in __init__
    until = datetime.date()
TypeError: descriptor 'date' of 'datetime.datetime' object needs an argument
```
when trying to edit this event:
```
BEGIN:VCALENDAR
VERSION:2.0
PRODID:-//PIMUTILS.ORG//NONSGML khal / icalendar //EN
BEGIN:VTIMEZONE
TZID:America/Chicago
BEGIN:DAYLIGHT
DTSTART;VALUE=DATE-TIME:20170312T030000
TZNAME:CDT
TZOFFSETFROM:-0600
TZOFFSETTO:-0500
END:DAYLIGHT
BEGIN:STANDARD
DTSTART;VALUE=DATE-TIME:20171105T010000
TZNAME:CST
TZOFFSETFROM:-0500
TZOFFSETTO:-0600
END:STANDARD
END:VTIMEZONE
BEGIN:VEVENT
SUMMARY:Kaushik
DTSTART;TZID=America/Chicago:20170830T100000
DTEND;TZID=America/Chicago:20170830T103000
DTSTAMP:20170823T160904Z
UID:8475c1ea-2a23-4046-8893-885a56416062
SEQUENCE:1
RRULE:FREQ=WEEKLY;UNTIL=20171229T160000Z
CREATED:20170823T160843Z
LAST-MODIFIED:20170823T160904Z
TRANSP:OPAQUE
X-MOZ-GENERATION:2
BEGIN:VALARM
ACTION:DISPLAY
DESCRIPTION:Default Mozilla Description
TRIGGER;VALUE=DURATION:-PT10M
END:VALARM
END:VEVENT
END:VCALENDAR
```
in ikhal v0.9.6 on Python 3.6.

**Comment by geier** [comment]
Hi and thank you for reporting this. I believe I just fixed this (in the new 0.9.x branch and in master). I'll try to get a 0.9.7 release including some other fixes out this weekend, otherwise you'll need to install from source.

**Comment by inducer** [comment]
Thanks for getting to this so quickly. I can confirm it's fixed on master.
